cogs/moderation.py

```diff
@@ -1,29 +1,57 @@
 import asyncio
+from collections import Counter
+import typing as t
+from .utils.formats import plural
 import textwrap
 
 import discord
 from discord import Color, Embed
 from discord.ext.commands import (
+    BadArgument,
     Bot,
     Cog,
     Context,
+    Converter,
     bot_has_permissions,
     command,
+    guild_only,
+    Greedy,
     has_permissions,
+    MemberConverter,
 )
 
 
-class Moderation(Cog):
-    """This cog provides moderation commands."""
+def can_execute_action(ctx: Context, user: discord.User, target: discord.User) -> bool:
+    return user.id == ctx.bot.owner_id or user == ctx.guild.owner or user.top_role > target.top_role
+
+
+class MemberNotFound(Exception):
+    pass
+
+
+async def resolve_member(self, guild: discord.Guild, member_id: int) -> discord.User:
+    user = guild.get_member(member_id)
+    if user is None:
+        if guild.chunked:
+            raise MemberNotFound()
+        try:
+            user = await guild.fetch_member(member_id)
+        except discord.NotFound:
+            raise MemberNotFound()
+    return user
```

Comment:
* Why is this taking `self` as an argument, this function isn't a part of any class.
* We should also use `bot.fetch_user(member_id)` as shown in the [docs](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html?highlight=fetch_user#discord.ext.commands.Bot.fetch_user) this is much better as it will find even the users which aren't connected to the guild, which can be helpful in case user which admins wanted to ban just disconnected. This also renders `guild` parameter irrelevant and we should accept `bot` instead.
* We should rename this function to `resolve_user` because of that

The final function may look something like this, but note that you also need to change all references to that function and this might not work so don't just paste it in, check it first.
```py
async def resolve_user(bot: Bot, member_id: int) -> discord.User:
    user = await bot.fetch_user(member_id)
    if not user:
        raise MemberNotFound
    return user
```